backend/src/main/java/org/sefglobal/core/academix/service/SubCategoryService.java

```diff
@@ -40,4 +40,19 @@ public SubCategoryService(SubCategoryRepository subCategoryRepository, CategoryR
         }
         return subCategoryRepository.getAllByCategory_Id(categoryId);
     }
+
+    /**
+     * Retrieves the subcategory with the requested id
+     * @param subCategoryId which is the id of the requested subCategory
+     * @return SubCategory object
+     * @throws ResourceNotFoundException if a subcategory with the requested id doesn't exist
+     */
+    public SubCategory getSubCategoryById(Long subCategoryId) throws ResourceNotFoundException {
+        if(!subCategoryRepository.existsById(subCategoryId)){
+            String msg = "Error, Sub Category by id:"+subCategoryId+" doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        return subCategoryRepository.getOne(subCategoryId);
```

Comment:
You're querying a category twice here by `subCategoryRepository.existsById(categoryId)` and `subCategoryRepository.getOne(categoryId)` which is unnecessary.
Instead, you can use `findById` method and return an `Optional` and then validate that.

Reply:
Created an `Optional<CustomCategory>` variable and validated it. But the `findById` method doesn't return a `CustomCategory`. So i created `findBy_Id` method in `CategoryRepository`